experimenter/experimenter/kinto/tasks.py

```diff
@@ -86,19 +87,32 @@ def nimbus_check_kinto_push_queue_by_collection(collection):
     handle_launching_experiments(applications, records)
     handle_updating_experiments(applications, records)
     handle_ending_experiments(applications, records)
-    handle_waiting_experiments(applications)
+    handle_waiting_experiments(applications, collection)
+
+    # NB: These are intentionally generator expressions so they can be lazily
+    # evaluated. Otherwise, we will do unnecessary queries for, e.g., the end
+    # queue and pause queue if we launch an experiment.
+    launch_queue = (
+        e
+        for e in NimbusExperiment.objects.launch_queue(applications)
+        if e.kinto_collection == collection
+    )
+    end_queue = (
+        e
+        for e in NimbusExperiment.objects.end_queue(applications)
+        if e.kinto_collection == collection
+    )
+    pause_queue = (
+        e
+        for e in NimbusExperiment.objects.update_queue(applications)
+        if e.kinto_collection == collection
+    )
```

Comment:
I think these manager methods are only used here, if we're always gonna filter by collection, should we just do that inside the manager methods?